src/extension/popup.ts

```diff
@@ -1,33 +1,77 @@
+    var startRecording = (<HTMLElement>document.getElementById("start"));
+    var stopRecording = (<HTMLElement>document.getElementById("stop"));
+    var replayLastSession = (<HTMLAnchorElement>document.getElementById("replayLastSession"));
+    var replayMenu = (<HTMLElement>document.getElementById("replayMenu"));
+    var date = (<HTMLElement>document.getElementById("date"));
+    var saveSession = (<HTMLAnchorElement>document.getElementById("save"));
+    var discardSession = (<HTMLAnchorElement>document.getElementById("discard"));
+    var inputName = (<HTMLElement>document.getElementById("nickname"));
     var showText = (<HTMLInputElement>document.getElementById("showText"));
     var showImages = (<HTMLInputElement>document.getElementById("showImages"));
     var showLines = (<HTMLInputElement>document.getElementById("showLines"));
-    var enabled = (<HTMLInputElement>document.getElementById("enabled"));
-    var replaySession = (<HTMLAnchorElement>document.getElementById("replaySession"));
     var menu = (<HTMLElement>document.getElementById("menu"));
 
     // Read from default storage
-    chrome.storage.sync.get({clarity: state}, function(items) {
+    chrome.storage.local.get({clarity: state}, function(items) {
         state = items.clarity;
         redraw(state);
-    }); 
+    });
+
+    function setDate() {
+        var now = new Date();
+        (<HTMLElement>date).innerHTML=now.getFullYear().toString();
+    }
 
     // Listen for changes
-    showText.addEventListener("click", toggle);
-    showImages.addEventListener("click", toggle);
-    showLines.addEventListener("click", toggle);
-    enabled.addEventListener("click", toggle);
-    replaySession.addEventListener("click", replay);
+    startRecording.addEventListener("click", toggleRecording);
+    stopRecording.addEventListener("click", toggleRecording);
+    replayLastSession.addEventListener("click", replay);
+    showText.addEventListener("click", toggleSettings);
+    showImages.addEventListener("click", toggleSettings);
+    showLines.addEventListener("clicl", toggleSettings);
 
     function replay() {
         chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
             var tab = tabs[0];
             chrome.tabs.create({ url: chrome.extension.getURL('clarity.html?tab=' + tab.id) });
         });
     }
-    function toggle(cb) {
+
+    function saveDiscardSession(cb) {      
+
+        if (cb.target.id == "save") {
+            // save session: future functionality?
+        }
+        else if (cb.target.id == "discard") {
+            // discard session: future functionality?
+        }
+    }
+
+    function toggleRecording(cb) {
+        state.recording = (cb.target.id == "start") ? true : false;
+        
+        if (!state.recording) {
+            // Any future functionality needed when stopped
+            state.uploaded = false;
```

Comment:
> state.uploaded = false; [](start = 11, length = 24)

does toggling recording to off do anything currently? I don't see any wireup that handles state.uploaded